security/cores.py

```diff
@@ -38,14 +31,59 @@ def display_as_code(value):
     return format_html('<code style="white-space:pre-wrap;">{}</code>', value) if value else value
 
 
-
 class DisplayRelatedObjectsMixin:
 
     @short_description(_('related objects'))
     def display_related_objects(self, obj, request):
         return render_model_objects_with_link(
             request,
-            [related_object.object for related_object in obj.related_objects.all() if related_object.object]
+            [related_object.object for related_object in obj.related_objects.exclude(
+                object_ct__in={
+                    ContentType.objects.get_for_model(model_class)
+                    for model_class in (CommandLog, InputLoggedRequest, OutputLoggedRequest,
+                                        CeleryTaskLog, CeleryTaskRunLog)
+                }
+            ) if related_object.object]
+        )
+
+    @short_description(_('source'))
+    def display_source(self, obj, request):
+        return render_model_objects_with_link(
+            request,
+            [related_object.object for related_object in obj.related_objects.filter(
+                object_ct__in={
+                    ContentType.objects.get_for_model(model_class)
+                    for model_class in (CommandLog, InputLoggedRequest, OutputLoggedRequest,
+                                        CeleryTaskLog, CeleryTaskRunLog)
+                }
```

Comment:
This is a static set, can you put it in helper function (you're using it twice here)